src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -0,0 +1,280 @@
+        {
+            _applicationHostOptions = applicationHostOptions;
+            _logger = loggerFactory?.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _httpClient = httpClient;
+            _secretManagerProvider = secretManagerProvider;
+            _configuration = configuration;
+            _hostIdProvider = hostIdProvider;
+        }
+
+        private CloudBlockBlob HashBlob
+        {
+            get
+            {
+                if (_hashBlob == null)
+                {
+                    string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+                    CloudStorageAccount account = null;
+                    if (!string.IsNullOrEmpty(storageConnectionString) &&
+                        CloudStorageAccount.TryParse(storageConnectionString, out account))
+                    {
+                        string hostId = _hostIdProvider.GetHostIdAsync(CancellationToken.None).GetAwaiter().GetResult();
+                        CloudBlobClient blobClient = account.CreateCloudBlobClient();
+                        var blobContainer = blobClient.GetContainerReference(ScriptConstants.AzureWebJobsHostsContainerName);
+                        string hashBlobPath = $"synctriggers/{hostId}/last";
+                        _hashBlob = blobContainer.GetBlockBlobReference(hashBlobPath);
+                    }
+                }
+                return _hashBlob;
+            }
+        }
+
+        public async Task<(bool success, string error)> TrySyncTriggersAsync(bool checkHash = false)
+        {
+            var payload = await GetSyncTriggersPayload();
+            string json = JsonConvert.SerializeObject(payload);
+
+            bool shouldSyncTriggers = true;
+            if (checkHash)
+            {
+                shouldSyncTriggers = await CheckAndUpdateHashAsync(json);
+            }
+
+            if (shouldSyncTriggers)
+            {
+                return await SetTriggersAsync(json);
+            }
+
+            return (false, null);
+        }
+
+        private async Task<bool> CheckAndUpdateHashAsync(string content)
+        {
+            try
+            {
+                if (HashBlob != null)
+                {
+                    // compute the current hash value
+                    string currentHash = null;
+                    using (var md5 = MD5.Create())
+                    {
+                        byte[] hash = md5.ComputeHash(Encoding.UTF8.GetBytes(content));
+                        currentHash = hash
+                            .Aggregate(new StringBuilder(), (a, b) => a.Append(b.ToString("x2")))
+                            .ToString();
+                    }
```

Comment:
I would avoid using MD5 in this code. Its considered unsafe in many contexts and as a result, code scanning tools catch it and report problems. While it would work fine for our purposes here, we can save our selves trouble by just using something like SHA256.